### PR TITLE
Feat/civ 4822 spec state flow test

### PR DESCRIPTION
**Pull request by filippo-possenti-version1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-4822

### Change description ###
Stateflow specified testing

StateFlowEngineSpecTest.java
CaseDataBuilderSpec.java

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by bhagyashreesharma90 — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            arguments(CaseDataBuilderSpec.builder().atStateClaimSubmitted2v1().build()),
+            arguments(CaseDataBuilderSpec.builder().atStateSpec1v1DefendantUnregisteredClaimSubmitted().build()),
+            arguments(CaseDataBuilderSpec.builder().atStateSpec2v1DefendantUnregisteredClaimSubmitted().build())
+        );
+    }
+
+    static Stream<Arguments> caseDataStreamTwoRespondentRepresentatives() {
+        return Stream.of(
+            arguments(CaseDataBuilderSpec.builder().atStateClaimSubmittedTwoRespondentDifferentSolicitorSpec().build()),
+            arguments(CaseDataBuilderSpec.builder().atStateSpec1v2Solicitor1UnregisteredSolicitor2Registered().build())
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:98]: could you be more specific on the method name - atStateSpec1v2Solicitor1UnregisteredSolicitor2Registered()

use instead - 
atStateSpec1v2Solicitor1UnregisteredSolicitor2Registered&Represented()

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        //When
+        StateFlow stateFlow = stateFlowEngine.evaluate(caseData);
+
+        // Then Claim will have SPEC_RPA_CONTINUOUS_FEED, NOTICE_OF_CHANGE, GENERAL_APPLICATION_ENABLED, CERTIFICATE_OF_SERVICE and RPA_CONTINUOUS_FEED
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.SPEC_RPA_CONTINUOUS_FEED.name(), false),
+            entry(FlowFlag.NOTICE_OF_CHANGE.name(), false),
+            entry(FlowFlag.GENERAL_APPLICATION_ENABLED.name(), false),
+            entry(FlowFlag.CERTIFICATE_OF_SERVICE.name(), false),
+            entry(FlowFlag.RPA_CONTINUOUS_FEED.name(), false)
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:132]: entry(FlowFlag.RPA_CONTINUOUS_FEED.name(), false) - this entry is not required as not used by spec claims

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        assertThat(stateFlow.getFlags()).contains(
+            entry(FlowFlag.ONE_RESPONDENT_REPRESENTATIVE.name(), true)
+        );
+        assertThat(stateFlow.getFlags()).hasSize(6);    // bonus: if this fails, a flag was added/removed but tests were not updated
+    }
+
+    @ParameterizedTest(name = "{index}: The state flow flags ONE_RESPONDENT_REPRESENTATIVE and " +
+        "TWO_RESPONDENT_REPRESENTATIVES are present and set to the correct values (for appropriate cases)")
+    @MethodSource("caseDataStreamTwoRespondentRepresentatives")
+    void shouldReturnClaimSubmitted_whenCaseDataAtStateClaimSubmitted1v2DifferentSolicitorSpecified(CaseData caseData) {
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:152]: be more specific in the name like whenCaseDataAtStateClaimSubmitted1v2DifferentSol1repAnd1unreg

Comment (akriti-hmcts) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:152]: no need of using specified as the test class is for specified only

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java (hunk 1 above)
Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:98]: done

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java (hunk 2 above)
Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:132]: done

File: src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java (hunk 3 above)
Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java:152]: This test is for two separate 1v2 scenarios //AC 3 (1V2 different defendant solicitor) and //AC 11 1v2 def 1 represented solicitor unregistered, and def 2 solicitor registered

Comment (drummondjm) [reply src/test/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngineSpecTest.java]: Updated name as it's for testing the flag on the above scenarios (ac3 and ac11)
shouldReturnFlags_whenAtStateClaimSubmitted1v2DiffSol_OrSolicitor1UnrepAndSolicitor2Registered

**Review by akriti-hmcts — APPROVED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by nagaraju-deekonda — APPROVED** [review]

**Pull request merged**